### PR TITLE
Add Webext Feature flag

### PR DESCRIPTION
**Pull request by strseb**
## Description

The Extension itself cannot know on which plattform it is on, and if that is supported. 
So let's have the client send a signal whether it supports "webextension" 

Also while at it, i consolidated a few `FeatureListCallback_X` with a consteval func, so in the end the compiler will evaluate to a simple `()=>true|false` lambda same as before but i think it looks neat to not change to a diffrent file to check when a feature is enabled 😅

### CODE REVIEW

**Comment by strseb** [comment]
Okay gcc/clang 12 hates it.

File: src/feature/featurelist.h [hunk 1]
```diff
@@ -38,14 +41,21 @@ FEATURE(annualUpgrade,         // Feature ID
         FeatureCallback_true,  // Can be flipped on
         FeatureCallback_true,  // Can be flipped off
         QStringList(),         // feature dependencies
-        FeatureCallback_annualUpgrade)
+        enableForPlatform({
+            .windows = true,
+            .macos = true,
+            .linux = true,
```
Comment (mcleinman) [inline src/feature/featurelist.h:47]: To match prior, it should have WASM here.

File: src/feature/featurelist.h [hunk 2]
```diff
@@ -214,18 +225,37 @@ FEATURE(unsecuredNetworkNotification,      // Feature ID
...
+
+FEATURE(webExtension,      // Feature ID
+        "webExtension",    // Feature name
+        enableForPlatform({// Can be flipped on
+                           .windows = true,
+                           .macos = true,
+                           .linux = true}),
+        FeatureCallback_true,  // Can be flipped off
+        QStringList(),         // feature dependencies
+        enableForPlatform({    // default value
```
Comment (mcleinman) [inline src/feature/featurelist.h:260]: This means "default true for windows, default false elsewhere", right? The naming `enableForPlatform` starts to feel a little weird - perhaps `platformBoolean` or something like that?

**Review by mcleinman — APPROVED** [review]
Thanks for this. A couple things that you could change, but nothing that blocks this, given current timelines.

File: src/feature/featurelist.h (hunk 1 above)
Comment (strseb) [reply src/feature/featurelist.h]: thanks! fixed.

**Pull request merged**